src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlConfiguration.java

```diff
@@ -316,6 +333,25 @@ public boolean finishOptionSettings() {
                 }
             }
         }
+        additionalScripts = options.additionalScripts().stream().map(sf -> {
+            DocFile file = DocFile.createFileForInput(this, sf);
+            boolean isModule = sf.toLowerCase(Locale.ROOT).endsWith(".mjs");
+            if (!isModule) {
+                // Regex to detect JavaScript modules
+                Pattern modulePattern = Pattern.compile("""
+                        (?:^|[;}])\\s*(?:\
+                        import\\s*["']|\
+                        import[\\s{*][^()]*from\\s*["']|\
+                        export(?:\\s+(?:let|const|function|class|var|default|async)|\\s*[{*]))""");
+                try (InputStream in = file.openInputStream();
+                     BufferedReader reader = new BufferedReader(new InputStreamReader(in))) {
+                    isModule = reader.lines().anyMatch(s -> modulePattern.matcher(s).find());
+                } catch (DocFileIOException | IOException e) {
+                    // Errors are handled when copying resources
+                }
+            }
+            return new JavaScriptFile(DocPath.create(file.getName()), isModule);
+        }).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
I realize it is not a change in this PR, but is there a reason to use `Collectors.toCollection(ArrayList::new)` instead of `Collectors.toList()`